### PR TITLE
[record] update __doc__

### PR DESCRIPTION
**Pull request by alangenfeld**
so apidocs work

## How I Tested These Changes

added test

`make apidoc-build`
` make next-watch-build`
load docs and ensure api entry for `DbtProject`

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#22952** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22952?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by maximearmstrong — APPROVED** [review]

**Review by erinkcochran87 — APPROVED** [review]

**Review by maximearmstrong — DISMISSED** [review]
Could it be missing the members? Looking at the documentation preview, the docstring for the class is back, but methods are missing.

**Comment by alangenfeld** [comment]
I dont see members on https://release-1-7-10.dagster.dagster-docs.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DbtProject

I think `:members:` has to be used on the autodoc entry

**Comment by maximearmstrong** [comment]
> I dont see members on https://release-1-7-10.dagster.dagster-docs.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DbtProject
> 
> I think `:members:` has to be used on the autodoc entry

There were no members at the time - `prepare_if_dev()` was introduced later. 

DagsterDbtTranslator has its members in [the docs](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DagsterDbtTranslator), and is not using `:members:` in the autodoc entry. I believe methods marked with the `@public` decorator are added [to the docs automatically](https://github.com/dagster-io/dagster/blob/36968cc4598b7e5a9fedb4f26a93fbde93675cd0/python_modules/dagster/dagster/_annotations.py#L33-L40).

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 1]
```diff
@@ -137,8 +137,8 @@ class DbtProject(IHaveNew):
 
     Using this helps achieve a setup where the dbt manifest file
     and dbt dependencies are available and up-to-date:
-        * during development, pull the dependencies and reload the manifest at run time to pick up any changes.
-        * when deployed, expect a manifest that was created at build time to reduce start-up time.
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:141]: this indentation from https://github.com/dagster-io/dagster/pull/22953  breaks the docs build

File: docs/sphinx/_ext/dagster-sphinx/dagster_sphinx/__init__.py [hunk 2]
```diff
@@ -86,6 +87,11 @@ class DagsterClassDocumenter(ClassDocumenter):
     objtype = "class"
 
     def get_object_members(self, want_all: bool) -> Tuple[bool, ObjectMembers]:
+        # the @record transform creates a new outer class, so redirect
+        # sphinx to target the original class for scraping members out of __dict__
+        if is_record(self.object):
+            self.object = get_original_class(self.object)
```
Comment (alangenfeld) [inline docs/sphinx/_ext/dagster-sphinx/dagster_sphinx/__init__.py:93]: [1]

**Review by alangenfeld — COMMENTED** [review]
ok [1] fixes the `:members:` issue

**Review by maximearmstrong — APPROVED** [review]

**Pull request merged**